### PR TITLE
feat: adds component to contain `LogoPtr` image resolution and fallback process

### PR DESCRIPTION
**Pull request by gdixon**
This PR abstracts `ptrToURI()` calls to a component (`LogoPtrImage`) - this component should be responsible for the `logoPtr` handling and implementation details.

--

Fixes: #557

### CODE REVIEW

**Review by jjwoz — APPROVED** [review]
lgtm - ran through the application and everything is working as expected from my point of view

File: app/src/utils/data/contributions.ts [hunk 1]
```diff
@@ -445,7 +445,7 @@ export function filterContributionGrantData(
     if (grantData && grantMetaData && Object.keys(grantMetaData).length) {
       const metaDataVersionId = metadataId(grantData.metaPtr);
       const { logoPtr, name } = grantMetaData[metaDataVersionId];
-      grantLogo = ptrToURI(logoPtr, grantLogo);
+      grantLogo = (logoPtr && logoPtr.protocol === 1 ? ptrToURI(logoPtr) : false) || grantLogo;
```
Comment (mds1) [inline app/src/utils/data/contributions.ts:448]: This should also work right? `ptrToUri` will return null if the protocol is not 1
```suggestion
      grantLogo = ptrToURI(logoPtr) || grantLogo;
```

**Review by mds1 — APPROVED** [review]
Code LGTM, just one small comment

File: app/src/utils/data/contributions.ts (hunk 1 above)
Comment (gdixon) [reply app/src/utils/data/contributions.ts:448]: I think you're right! Before I had this setup as `IpfsImg` and figured the component didn't need to be so specific, this is a remnant! I'll try it out and get it committed, thanks!

**Pull request merged**